python/plugins/processing/tests/AlgorithmsTest.py

```diff
@@ -0,0 +1,168 @@
+        """
+        params = self.load_params(defs['params'])
+
+        alg = processing.Processing.getAlgorithm(defs['algorithm']).getCopy()
+        for param in zip(alg.parameters, params):
+            param[0].setValue(param[1])
+
+        for r, p in defs['results'].iteritems():
+            alg.setOutputValue(r, self.load_result_param(p))
+
+        self.assertTrue(AlgorithmExecutor.runalg(alg))
+        print(alg.getAsCommand())
+        self.check_results(alg.getOutputValuesAsDictionary(), defs['results'])
+
+    def load_params(self, params):
+        """
+        Loads an array of parameters
+        """
+        return [self.load_param(p) for p in params]
+
+    def load_param(self, param):
+        """
+        Loads a parameter. If it's not a map, the parameter will be returned as-is. If it is a map, it will process the
+        parameter based on its key `type` and return the appropriate parameter to pass to the algorithm.
+        """
+        try:
+            if param['type'] == 'vector' or param['type'] == 'raster':
+                return self.load_layer(param)
+        except TypeError:
+            # No type specified, use whatever is there
+            return param
+
+        raise KeyError("Unknown type '{}' specified for parameter '{}'".format(param['type'], param['name']))
+
+    def load_result_param(self, param):
+        """
+        Lodas a result parameter. Creates a temporary destination where the result should go to and returns this location
+        so it can be sent to the algorithm as parameter.
+        """
+        if param['type'] == 'vector':
+            outdir = tempfile.mkdtemp()
+            basename = os.path.basename(param['name'])
+            filepath = os.path.join(outdir, basename)
+            return filepath
+
+        raise KeyError("Unknown type '{}' specified for parameter '{}'".format(param['type'], param['name']))
+
+    def load_layer(self, param):
+        """
+        Loads a layer which was specified as parameter.
+        """
+        if param['location'] == 'qgs':
+            prefix = unitTestDataPath()
+        elif param['location'] == 'proc':
+            prefix = processingTestDataPath()
+        else:
+            prefix = ''
+
+        filepath = os.path.join(prefix, param['name'])
+
+        if param['type'] == 'vector':
+            lyr = QgsVectorLayer(filepath, param['name'], 'ogr')
+        elif param['type'] == 'raster':
+            lyr = QgsRasterLayer(filepath, param['name'], 'ogr')
+
+        self.assertTrue(lyr.isValid(), 'Could not load layer "{}"'.format(filepath))
+        QgsMapLayerRegistry.instance().addMapLayer(lyr)
+        return lyr
+
+    def check_results(self, results, expected):
+        """
+        Checks if result produced by an algorithm matches with the expected specification.
+        """
+        for id, expected_result in expected.iteritems():
+            if expected_result['type'] == 'vector':
```

Comment:
Yes, it's on the todo list to add more (see first comment in this PR). This PR is mainly meant for early feedback about the general approach.